src/algorithms/excitation/TM_excitations.jl

```diff
@@ -0,0 +1,26 @@
+function correlation_length(state::InfiniteMPS; otherstate=state, tol = 1e-8, exi_space = ComplexSpace(1), tol_angle = 0.1, triv=true, max_sinvals = 20)
```

Comment:
So why is this argument called `max_sinvals`? It's not singular values right? It's eigenvalues of the transfer matrix. Maybe more generic name like `howmany` would be more appropriate? 

Instead of `exi_space`, I would rather specify a sector (and call the argument `sector::Sector = Trivial()`). The corresponding exi_space to be used is then `exi_space = ℂ[sector=>1]`. This also works for `sector = Trivial()` and you can even use the trivial case `exi_space = ℂ^1` when the spacetype is not `ComplexSpace`, by replacing `space(ALs2[1],1)*exi_space` with `space(ALs2[1],1) ⊠ exi_space` (the deligne tensor product at work). 

The reason not to directly give `exi_space` is that, if e.g. a user gives  `exi_space =  ℂ^2`, this will just duplicate every eigenvalue, cause you have a transfer matrix acting on index 1 and 3 of your eigenvector, and and identity map acting on the second index (the `exi_space` index), so essentially you are computing the eigenvalues of transfer matrix kronecker product with an identity matrix.

Reply:
this would also be better in the quasiparticle excitation code